decoder/semantic_tokens_test.go

```diff
@@ -836,3 +852,204 @@ resource "vault_auth_backend" "blah" {
+						SemanticTokenModifiers: lang.SemanticTokenModifiers{
+							lang.TokenModifierDependent,
+						},
+					},
+					{Name: "name"},
+				},
+			},
+		},
+	}
+
+	testCfg := []byte(`
+resource "vault_auth_backend" "blah" {
+	count = 1
+  cpu_count = count.index
+}
+`)
+
+	f, pDiags := hclsyntax.ParseConfig(testCfg, "test.tf", hcl.InitialPos)
+	if len(pDiags) > 0 {
+		t.Fatal(pDiags)
+	}
+
+	d := testPathDecoder(t, &PathContext{
+		Schema: bodySchema,
+		Files: map[string]*hcl.File{
+			"test.tf": f,
+		},
+	})
+
+	ctx := context.Background()
+
+	tokens, err := d.SemanticTokensInFile(ctx, "test.tf")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedTokens := []lang.SemanticToken{
+		{ // resource
+			Type:      lang.TokenBlockType,
+			Modifiers: []lang.SemanticTokenModifier{},
+			Range: hcl.Range{
+				Filename: "test.tf",
+				Start: hcl.Pos{
+					Line:   2,
+					Column: 1,
+					Byte:   1,
+				},
+				End: hcl.Pos{
+					Line:   2,
+					Column: 9,
+					Byte:   9,
+				},
+			},
+		},
+		{ // vault_auth_backend
+			Type: lang.TokenBlockLabel,
+			Modifiers: []lang.SemanticTokenModifier{
+				lang.TokenModifierDependent,
+			},
+			Range: hcl.Range{
+				Filename: "test.tf",
+				Start: hcl.Pos{
+					Line:   2,
+					Column: 10,
+					Byte:   10,
+				},
+				End: hcl.Pos{
+					Line:   2,
+					Column: 30,
+					Byte:   30,
+				},
+			},
+		},
+		{ // blah
+			Type:      lang.TokenBlockLabel,
+			Modifiers: []lang.SemanticTokenModifier{},
+			Range: hcl.Range{
+				Filename: "test.tf",
+				Start: hcl.Pos{
+					Line:   2,
+					Column: 31,
+					Byte:   31,
+				},
+				End: hcl.Pos{
+					Line:   2,
+					Column: 37,
+					Byte:   37,
+				},
+			},
+		},
+		{ // count
+			Type:      lang.TokenAttrName,
+			Modifiers: lang.SemanticTokenModifiers{},
+			Range: hcl.Range{
+				Filename: "test.tf",
+				Start: hcl.Pos{
+					Line:   3,
+					Column: 2,
+					Byte:   41,
+				},
+				End: hcl.Pos{
+					Line:   3,
+					Column: 7,
+					Byte:   46,
+				},
+			},
+		},
+		{ // 1
+			Type:      lang.TokenNumber,
+			Modifiers: lang.SemanticTokenModifiers{},
+			Range: hcl.Range{
+				Filename: "test.tf",
+				Start: hcl.Pos{
+					Line:   3,
+					Column: 10,
+					Byte:   49,
+				},
+				End: hcl.Pos{
+					Line:   3,
+					Column: 11,
+					Byte:   50,
+				},
+			},
+		},
+		{ // cpu_count
+			Type:      lang.TokenAttrName,
+			Modifiers: lang.SemanticTokenModifiers{},
+			Range: hcl.Range{
+				Filename: "test.tf",
+				Start: hcl.Pos{
+					Line:   4,
+					Column: 3,
+					Byte:   53,
+				},
+				End: hcl.Pos{
+					Line:   4,
+					Column: 12,
+					Byte:   62,
+				},
+			},
+		},
+		{ // count.
+			Type:      lang.TokenTraversalStep,
+			Modifiers: lang.SemanticTokenModifiers{},
+			Range: hcl.Range{
+				Filename: "test.tf",
+				Start: hcl.Pos{
+					Line:   4,
+					Column: 15,
+					Byte:   65,
+				},
+				End: hcl.Pos{
+					Line:   4,
+					Column: 20,
+					Byte:   70,
+				},
+			},
+		},
+		{ // .index
```

Comment:
The comment reflects the current implementation correctly but as mentioned, we typically only report the segments as segments, not the separator.